metadata/metadata.go

```diff
@@ -54,6 +54,67 @@ func AddLabels(obj v1.Object, entries map[string]string) error {
 	return nil
 }
 
+// CopyAnnotationsByPrefix copies all annotations from a source object to a destination object where the key matches
+// the specified sourcePrefix.
+func CopyAnnotationsByPrefix(source, destination v1.Object, prefix string) error {
+	if source == nil || destination == nil {
+		return errors.New("object cannot be nil")
+	}
+
+	if destination.GetAnnotations() == nil {
+		destination.SetAnnotations(make(map[string]string))
+	}
+	copyByPrefix(source.GetAnnotations(), destination.GetAnnotations(), prefix)
+
+	return nil
+}
+
+// CopyAnnotationsWithPrefixReplacement copies all annotations from a source object to a destination object where the
+// key matches the specified sourcePrefix.
+// The source prefix will be replaced with the destination prefix.
+func CopyAnnotationsWithPrefixReplacement(source, destination v1.Object, sourcePrefix, destinationPrefix string) error {
+	if source == nil || destination == nil {
+		return errors.New("object cannot be nil")
+	}
+
+	if destination.GetAnnotations() == nil {
+		destination.SetAnnotations(make(map[string]string))
+	}
+	copyWithPrefixReplacement(source.GetAnnotations(), destination.GetAnnotations(), sourcePrefix, destinationPrefix)
+
+	return nil
+}
+
+// CopyLabelsByPrefix copies all labels from a source object to a destination object where the key matches the
+// specified sourcePrefix.
+func CopyLabelsByPrefix(source, destination v1.Object, prefix string) error {
+	if source == nil || destination == nil {
+		return errors.New("object cannot be nil")
+	}
+
+	if destination.GetLabels() == nil {
+		destination.SetLabels(make(map[string]string))
+	}
+	copyByPrefix(source.GetLabels(), destination.GetLabels(), prefix)
+
+	return nil
+}
+
+// CopyLabelsWithPrefixReplacement copies all labels from a source object to a destination object where the key matches the specified sourcePrefix.
```

Comment:
Break line before `the specified` please.

Reply:
Updated!